spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/encoding/FeignAcceptGzipEncodingAutoConfiguration.java

```diff
@@ -37,8 +39,9 @@
 @Configuration
 @EnableConfigurationProperties(FeignClientEncodingProperties.class)
 @ConditionalOnClass(Feign.class)
-@ConditionalOnBean(ApacheHttpClient.class)
+@ConditionalOnBean(Client.class)
 @ConditionalOnProperty(value = "feign.compression.response.enabled", matchIfMissing = false)
+@ConditionalOnMissingBean(OkHttpClient.class)
```

Comment:
Why this change?

Reply:
@spencergibb yeah context would help, I added some to the description

Reply:
Makes sense. I think your context should be added as comments to the code.

Reply:
Done